server/src/controllers/course.ts

```diff
@@ -203,11 +212,12 @@ export const editCourse = async (
     try {
       await Course.update(
         {
-          courseCode: courseCode,
-          minCredits: minCredits,
-          maxCredits: maxCredits,
-          gradingScale: gradingScale,
-          languageOfInstruction: languageOfInstruction,
+          courseCode: courseCode ?? course.courseCode,
+          minCredits: minCredits ?? course.minCredits,
+          maxCredits: maxCredits ?? course.maxCredits,
+          gradingScale: gradingScale ?? course.gradingScale,
+          languageOfInstruction:
+            languageOfInstruction ?? course.languageOfInstruction,
```

Comment:
This actually does not change the functionality but I think this is more clear? It could also lead to bugs because if one of the values is null it will be ignored (in this case none of them can).